### DISCUSSION TITLE
transform = RandomNodeSplit(split='random') causes Colab to crash

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#4797] [Q&A] [answered]
Hi @rusty1s 

I am training node classification on Cora. Before i train the model I am performing Ricci Flow using this library - [Ricci Flow](https://graphriccicurvature.readthedocs.io/en/v0.5.1/tutorial.html#Ricci-flow) on the dataset. 
That is, convert PyG to NetworkX -

```Python
from torch_geometric.utils import to_networkx,from_networkx
G1 = to_networkx(data,node_attrs=['x'],graph_attrs=['y'])
print(type(datanx))
```
Compute Ricci Flow -

```Python
orf = OllivierRicci(G1, alpha=0.5, base=1, exp_power=0, proc=4, verbose="INFO")
orf.compute_ricci_flow(iterations=5)
G_rf = orf.G.copy()
```

Convert back to PyG

```Python
train_data = from_networkx(G_rf)
print(train_data)
Data(x=[2485, 1433], edge_index=[2, 10138], y=[2708], ricciCurvature=[2485], weight=[10138], edge_ricciCurvature=[10138], original_RC=[10138])
```
I want the train/test/val mask so Im using the transforms -

```Python
from torch_geometric.transforms import RandomNodeSplit
transform = RandomNodeSplit(split='random')
train_data = transform(train_data)
print(train_data)

```
My Colab crashes saying - Crashed for unknown reason. Is it somehow using up all the RAM thats causing the issue?  Thanks in advance :)

**Comment by rusty1s** [comment, marked as answer]
This is hard to tell, I am sorry. Where does it crash exactly? Is it in the `RandomNodeSplit` execution?

**Reply by AdarshMJ** [reply]
No actually its while applying the transforms -

`train_data = transform(train_data)`

**Reply by rusty1s** [reply]
Is it possible for you debug further, i.e, which statement inside `RandomNodeSplit`?

**Reply by AdarshMJ** [reply]
Hi I am sorry I am not sure I understand how to do that. Do you want me to check the source of` RandomNodeSplit`?

**Reply by rusty1s** [reply]
Basically doing the same as before: It would be great to see where the crash happens inside the computation in `RandomNodeSplit`, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/transforms/random_node_split.py#L72-L127). You might be able to do this via a debugger or via print debugging inside the source code.

**Reply by AdarshMJ** [reply]
Okay I can check this out. Thank you!
